packages/eth2.0-types/src/types/validator.ts

```diff
@@ -26,3 +27,61 @@ export interface SyncingStatus {
   // The estimated highest block, or current target block number
   highestBlock: uint64;
 }
+
+export interface Registry<T, T1> extends Array<T> {
```

Comment:
why not export class with registry prop and override constructor to initialize registry from given array (constructor param)

Reply:
array constructors are
```
    new(arrayLength?: number): any[];
    new <T>(arrayLength: number): T[];
    new <T>(...items: T[]): T[];
```

Reply:
The registry has 2 generic types: T and T1, T is a type like `Validator` and T1 is the type of field we want to index, it can be either number or string.
Do you want to drop T1? This implementation is generic anyway so that we can apply for similar scenarios like Validator.

Reply:
Array is an interface not class so I find no way to extend and have constructor for it like that... Please suggest one if you know.

Reply:
Maybe try something like this:
https://github.com/Microsoft/TypeScript/issues/13720#issuecomment-348487195

Also a good thread:
https://blog.simontest.net/extend-array-with-typescript-965cc1134b3